qiskit/dagcircuit/dagdependency.py

```diff
@@ -601,9 +601,14 @@ def _unknown_commutator(n):
     qarg1 = [qarg.index(q) for q in node1.qargs]
     qarg2 = [qarg.index(q) for q in node2.qargs]
 
-    id_op = Operator(np.eye(2 ** qbit_num))
+    dim = 2 ** qbit_num
+    id_op = np.reshape(np.eye(dim), (2, 2) * qbit_num)
 
-    op12 = id_op.compose(node1.op, qargs=qarg1).compose(node2.op, qargs=qarg2)
-    op21 = id_op.compose(node2.op, qargs=qarg2).compose(node1.op, qargs=qarg1)
+    op1 = np.reshape(node1.op, (2, 2) * len(qarg1))
+    op2 = np.reshape(node2.op, (2, 2) * len(qarg2))
```

Comment:
Does this work with `node1.op.to_matrix()`? When we did similar tuning in `Optimize1qGatesDecomposition` (in #5909) we found that relying on `__array__` and having numpy create an ndarray implicitly was slower than just constructing the numpy array explicitly and using that for the operation.

Reply:
yes, and it's indeed a bit faster.